### PR TITLE
fix broken dev branch

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- fix missing include
- fix unused variable errors due to constexpr if's

ci: full-compile

### CODE REVIEW

**Comment by ikbuibui** [comment]
Could you maybe include this change https://github.com/ComputationalRadiationPhysics/picongpu/pull/5147#discussion_r1784134264  into the PR as well?

File: include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel [hunk 1]
```diff
@@ -154,7 +154,8 @@ namespace picongpu::particles::atomicPhysics::kernel
 
             auto forEachAtomicStateAndBin
                 = pmacc::lockstep::makeForEach<T_numberAtomicStates * T_numberBins, T_Worker>(worker);
-            auto forEachAtomicState = pmacc::lockstep::makeForEach<T_numberAtomicStates, T_Worker>(worker);
+            [[maybe_unused]] auto forEachAtomicState
+                = pmacc::lockstep::makeForEach<T_numberAtomicStates, T_Worker>(worker);
```
Comment (chillenzer) [inline include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel]:
```suggestion
            auto forEachAtomicState
                = pmacc::lockstep::makeForEach<T_numberAtomicStates, T_Worker>(worker);
```
and move it down to where it's actually used. As far as I can see that's a single place. And maybe do the same for the `forEachAtomicStateAndBin` above.

**Review by chillenzer — DISMISSED** [review]
Not blocking.

File: include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel (hunk 1 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel]: it is used twice in two `if constexpr` branches but would like avoid creating the for each twice.

**Comment by psychocoderHPC** [comment]
> Could you maybe include this change [#5147 (comment)](https://github.com/ComputationalRadiationPhysics/picongpu/pull/5147#discussion_r1784134264) into the PR as well?

No, I try to avoid mixing it. There is a high change that the header checks will fail if I remove it. Most likely there is a reason why the include was added there. If there is an include too much we should remove it in a seperat PR. It could also be that this is indirect fixing a missing include somewhere.

File: include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel (hunk 1 above)
Comment (chillenzer) [reply include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel]: Doing "Click `...`" -> "View File" -> "Ctrl-F" -> "`forEachAtomicState`" -> "Enter" yields exactly four results for me: One definition and one usage per variable. Are you aware that these are two different variables?

Comment (psychocoderHPC) [reply include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel]: I missed it, the names are so equal that I oversaw it. 

fixed it

File: include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel [hunk 2]
```diff
@@ -154,7 +154,6 @@ namespace picongpu::particles::atomicPhysics::kernel
 
             auto forEachAtomicStateAndBin
                 = pmacc::lockstep::makeForEach<T_numberAtomicStates * T_numberBins, T_Worker>(worker);
```
Comment (chillenzer) [inline include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel:156]: Sorry for being pedantic but the same applies to this one. It should also move into the `if constexpr`.

**Review by chillenzer — CHANGES_REQUESTED** [review]

File: include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel (hunk 2 above)
Comment (psychocoderHPC) [reply include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel:156]: Yes, you are right possible to change in a separate PR. I only react on broken tests in the CI I have not done a code refactoring.

Comment (chillenzer) [reply include/picongpu/particles/atomicPhysics/kernel/FillLocalRateCache_BoundBound.kernel:156]: Okay.

**Review by chillenzer — APPROVED** [review]

**Pull request merged**